### PR TITLE
Parse most of the vector-like types during XAML compilation.

### PR DESCRIPTION
**Pull request by MarchingCube**
## What does the pull request do?

Adds support for parsing common vector like types during XAML compilation time. Currently it supports:
- `Thickness`
- `Point`
- `Size`
- `Vector`
- `CornerRadius`
- `Matrix`

## What is the current behavior?

XAML compiler simply outputs `Parse` call which has runtime overhead and requires allocations.

## What is the updated/expected behavior with this PR?

XAML compiler outputs direct constructor calls that create instances directly from already parsed data.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by MarchingCube** [comment]
I have lots of `#if !BUILDTASK` since currently many projects are referencing build task project which is causing duplicate class signatures.

File: src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AstNodes/AvaloniaXamlIlVectorLikeConstantAstNode.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+namespace Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.AstNodes
+{
+    class AvaloniaXamlIlVectorLikeConstantAstNode : XamlAstNode, IXamlAstValueNode, IXamlAstILEmitableNode
+    {
+        private readonly IXamlConstructor _constructor;
+        private readonly double[] _values;
+
+        public AvaloniaXamlIlVectorLikeConstantAstNode(IXamlLineInfo lineInfo, IXamlType type, IXamlConstructor constructor, double[] values) : base(lineInfo)
+        {
+            _constructor = constructor;
```
Comment (kekekeks) [inline src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AstNodes/AvaloniaXamlIlVectorLikeConstantAstNode.cs:15]: Validate the ctor signature, better do it here than by examining generated MSIL manually later

File: src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlWellKnownTypes.cs [hunk 2]
```diff
@@ -52,7 +53,18 @@ class AvaloniaXamlIlWellKnownTypes
         public IXamlType Uri { get; }
         public IXamlType FontFamily { get; }
         public IXamlConstructor FontFamilyConstructorUriName { get; }
-        
+        public IXamlType Thickness { get; }
+        public IXamlConstructor ThicknessFullConstructorName { get; }
```
Comment (kekekeks) [inline src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlWellKnownTypes.cs:57]: The `Name` in `FontFamilyConstructorUriName` wasn't a suffix, it was referring to `FontFamily(Uri uri, string name)` signature.

**Review by kekekeks — CHANGES_REQUESTED** [review]

File: src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlWellKnownTypes.cs (hunk 2 above)
Comment (MarchingCube) [reply src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlWellKnownTypes.cs]: Right, I guess I've copy pasted without thinking much about name here :)

**Review by kekekeks — APPROVED** [review]

**Pull request merged**